Feature/company users bulk rest api

daniel-rose opened the pull request.

Commit:
add base files

Commit:
add transfer xml and constants class

Commit:
add Basic structure for the Glue implementation (work in progress)

julianhyatt committed:
Merge branch 'main' into feature/company-users-bulk-rest-api

julianhyatt committed:
Added zed client
Added permission check
some refactoring

daniel-rose commented on bundles/company-users-bulk-rest-api/composer.json:
```diff
@@ -0,0 +1,42 @@
+    "spryker/company-user": "^1.0.0 || ^2.0.0"
+  },
+  "require-dev": {
+    "fond-of-codeception/spryker": "^1.0 || ^2.0",
```
```suggestion
    "fond-of-codeception/spryker": "*",
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiClient.php:
```diff
@@ -0,0 +1,23 @@
+     * @param \Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer $restCompanyUsersBulkRequestTransfer
+     * @return \Generated\Shared\Transfer\RestCompanyUsersBulkResponseTransfer
+     */
+    public function bulkProcess(
```
The name is perhaps not quite accurate.

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Glue/CompanyUsersBulkRestApi/Plugin/GlueApplicationExtension/CompanyUsersBulkResourceRoutePlugin.php:
```diff
@@ -0,0 +1,46 @@
+     */
+    public function configure(ResourceRouteCollectionInterface $resourceRouteCollection): ResourceRouteCollectionInterface
+    {
+        return $resourceRouteCollection->addPost('post');
```
Maybe you can use an constant for the string `post`

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Glue/CompanyUsersBulkRestApi/Processor/Mapper/RestCompanyUsersBulkRequestMapper.php:
```diff
@@ -0,0 +1,53 @@
+     */
+    public function createRequest(
+        RestRequestInterface                           $restRequest,
+        ?RestCompanyUsersBulkRequestAttributesTransfer $restCompanyUsersBulkRequestAttributesTransfer = null
```
PHPCS?

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Glue/CompanyUsersBulkRestApi/Processor/Mapper/RestCompanyUsersBulkRequestMapperInterface.php:
```diff
@@ -0,0 +1,20 @@
+     */
+    public function createRequest(
+        RestRequestInterface                           $restRequest,
+        ?RestCompanyUsersBulkRequestAttributesTransfer $restCompanyUsersBulkRequestAttributesTransfer = null
```
PHPCS?

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Shared/CompanyUsersBulkRestApi/Transfer/company_users_bulk_rest_api.transfer.xml:
```diff
@@ -0,0 +1,37 @@
+
+    <transfer name="RestCompanyUsersBulkResponse">
+        <property name="isSuccessful" type="bool"/>
+        <property name="error" type="string"/>
+        <property name="code" type="int"/>
```
Maybe you can use an error message transfer.

daniel-rose replied:
Note that the message must be set as a glossary key. Later you need the glossary storage client for the correct translation.

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Manager/BulkManager.php:
```diff
@@ -0,0 +1,42 @@
+use Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer;
+use Generated\Shared\Transfer\RestCompanyUsersBulkResponseTransfer;
+
+class BulkManager implements BulkManagerInterface
```
Maybe your naming is not clearly.

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Permission/PermissionChecker.php:
```diff
@@ -0,0 +1,36 @@
+     * @param \Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer $restCompanyUsersBulkRequestTransfer
+     * @return bool
+     */
+    public function checkPermission(
```
```suggestion
    public function check(
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Permission/PermissionCheckerInterface.php:
```diff
@@ -0,0 +1,16 @@
+     * @param \Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer $restCompanyUsersBulkRequestTransfer
+     * @return bool
+     */
+    public function checkPermission(
```
```suggestion
    public function check(
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Persistence/CompanyUsersBulkRestApiPersistenceFactory.php:
```diff
@@ -0,0 +1,40 @@
+
+class CompanyUsersBulkRestApiPersistenceFactory extends AbstractPersistenceFactory
+{
+    /**
+     * @return \Orm\Zed\Customer\Persistence\SpyCustomerQuery
+     */
+    public function getCustomerQuery(): SpyCustomerQuery
+    {
+        return $this->getProvidedDependency(CompanyUsersBulkRestApiDependencyProvider::QUERY_SPY_CUSTOMER);
+    }
```
Do you need this?

julianhyatt committed:
Update bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Permission/PermissionChecker.php

Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

julianhyatt committed:
Update bundles/company-users-bulk-rest-api/composer.json

Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

julianhyatt committed:
Update bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Permission/PermissionCheckerInterface.php

Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

julianhyatt committed:
Add plugin logic for data enrichment
Add Event for bulk creation
Add extension bundle for plugin interfaces

daniel-rose commented on bundles/company-users-bulk-rest-api-extension/src/FondOfOryx/Zed/CompanyUsersBulkRestApiExtension/Dependency/Plugin/CompanyUsersBulkItemPostEnrichmentPluginInterface.php:
```diff
@@ -0,0 +1,16 @@
+
+use Generated\Shared\Transfer\CompanyUsersBulkPreparationTransfer;
+
+interface CompanyUsersBulkItemPostEnrichmentPluginInterface
```
```suggestion
interface CompanyUsersBulkItemPostEnricherPluginInterface
```

OR

```suggestion
interface CompanyUsersBulkItemPostExpanderPluginInterface
```

daniel-rose commented on bundles/company-users-bulk-rest-api-extension/src/FondOfOryx/Zed/CompanyUsersBulkRestApiExtension/Dependency/Plugin/CompanyUsersBulkItemPostEnrichmentPluginInterface.php:
```diff
@@ -0,0 +1,16 @@
+     * @param \Generated\Shared\Transfer\CompanyUsersBulkPreparationTransfer $companyUsersBulkPreparationTransfer
+     * @return \Generated\Shared\Transfer\CompanyUsersBulkPreparationTransfer
+     */
+    public function postEnrichment(
```
```suggestion
    public function postEnrich(
```

OR

```suggestion
    public function postExpand(
```

daniel-rose commented on bundles/company-users-bulk-rest-api-extension/src/FondOfOryx/Zed/CompanyUsersBulkRestApiExtension/Dependency/Plugin/CompanyUsersBulkItemPreEnrichmentPluginInterface.php:
```diff
@@ -0,0 +1,16 @@
+     * @param \Generated\Shared\Transfer\CompanyUsersBulkPreparationTransfer $companyUsersBulkPreparationTransfer
+     * @return \Generated\Shared\Transfer\CompanyUsersBulkPreparationTransfer
+     */
+    public function preEnrichment(
```
```suggestion
    public function preEnrich(
```

OR

```suggestion
    public function preExpand(
```

daniel-rose commented on bundles/company-users-bulk-rest-api-extension/src/FondOfOryx/Zed/CompanyUsersBulkRestApiExtension/Dependency/Plugin/CompanyUsersBulkItemPreEnrichmentPluginInterface.php:
```diff
@@ -0,0 +1,16 @@
+
+use Generated\Shared\Transfer\CompanyUsersBulkPreparationTransfer;
+
+interface CompanyUsersBulkItemPreEnrichmentPluginInterface
```
```suggestion
interface CompanyUsersBulkItemPreEnricherPluginInterface
```

OR

```suggestion
interface CompanyUsersBulkItemPreExpanderPluginInterface
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Manager/BulkManager.php:
```diff
@@ -0,0 +1,129 @@
+//            return $this->createEmptyResponseTransfer()
+//                ->setCode(CompanyUsersBulkRestApiConstants::ERROR_CODE_PERMISSION_DENIED)
+//                ->setError(CompanyUsersBulkRestApiConstants::ERROR_MESSAGE_MISSING_PERMISSION);
+//        }
```
Remove unused code.

julianhyatt replied:
Its not unused, only uncommented for testing without permission check :)

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Communication/Plugin/CustomerResolverPreEnrichmentPlugin.php:
```diff
@@ -0,0 +1,51 @@
+/**
+ * @method \FondOfOryx\Zed\CompanyUsersBulkRestApi\Persistence\CompanyUsersBulkRestApiRepositoryInterface getRepository()
+ */
+class CustomerResolverPreEnrichmentPlugin extends AbstractPlugin implements CompanyUsersBulkItemPreEnrichmentPluginInterface
```
```suggestion
class CustomerResolverPreEnricherPlugin extends AbstractPlugin implements CompanyUsersBulkItemPreEnricherPluginInterface
```

OR

```suggestion
class CustomerResolverPreExpanderPlugin extends AbstractPlugin implements CompanyUsersBulkItemPreExpanderPluginInterface
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiDependencyProvider.php:
```diff
@@ -0,0 +1,194 @@
+        $container = $this->addSpyCompanyUserQuery($container);
+        $container = $this->addSpyCustomerQuery($container);
+        $container = $this->addPostEnrichmentPlugins($container);
+        $container = $this->addPreEnrichmentPlugins($container);
```
Naming. See above.

daniel-rose reviewed (commented):
Please start with the unit tests @julianzimmermann ...

julianhyatt committed:
Added new interfaces for plugins
Implemet bulk create and delete for company user

julianhyatt committed:
fix codestyles
add test directories

julianhyatt committed:
add missing requirements
fix ci stuff

julianhyatt committed:
some refactoring
remove pre and post enrichment plugins und functionality
add debtor connector bundle
extract debtor stuff to new bundle
introduce new expander plugins

julianhyatt committed:
add github actions
add readme

julianhyatt committed:
remove unused plugin interfaces
fix phpcs

julianhyatt committed:
add unit tests
fix php code styles
add missing requirements

julianhyatt committed:
Merge branch 'main' into feature/company-users-bulk-rest-api

julianhyatt committed:
fix phpstan

julianhyatt committed:
remove local stuff

julianhyatt committed:
add tests to main codeception.yml

julianhyatt committed:
wip

julianhyatt committed:
wip

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/src/FondOfOryx/Shared/CompanyUsersBulkRestApiDebtorConnector/Transfer/company_users_bulk_rest_api_debtor_connector.transfer.xml:
Not complete. You more then `RestCompanyUsersBulkItemCompany`.

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/composer.json:
```diff
@@ -0,0 +1,48 @@
+  "require": {
+    "php": ">=8.0",
+    "fond-of-oryx/company-users-bulk-rest-api": "^1.0.0",
+    "fond-of-spryker/navision-company": "dev-master",
```
Is that even needed?

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/composer.json:
```diff
@@ -0,0 +1,48 @@
+    "fond-of-oryx/company-users-bulk-rest-api": "^1.0.0",
+    "fond-of-spryker/navision-company": "dev-master",
+    "spryker/company-business-unit": "^1.0.0 || ^2.0.0",
+    "spryker/company-user": "^2.0.0",
```
Is version `1.0.0` not allowed? And Why?

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/src/FondOfOryx/Zed/CompanyUsersBulkRestApiDebtorConnector/Communication/Plugin/CompanyUsersBulkRestApi/CompanyDebtorNumberResolverPreDataExpanderPlugin.php:
```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApiDebtorConnector\Communication\Plugin\CompanyUsersBulkRestApi;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApiDebtorConnector\Communication\Plugin\CompanyUsersBulkRestApiExtension;
```

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/src/FondOfOryx/Zed/CompanyUsersBulkRestApiDebtorConnector/Communication/Plugin/CompanyUsersBulkRestApi/CompanyDebtorNumberResolverPreDataExpanderPlugin.php:
Put the hole logic to business layer please.

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/src/FondOfOryx/Zed/CompanyUsersBulkRestApiDebtorConnector/Persistence/CompanyUsersBulkRestApiDebtorConnectorRepository.php:
```diff
@@ -0,0 +1,47 @@
+                $companyTransfer->addCompanyRole((new CompanyRoleTransfer())->fromArray($role->toArray(), true));
+            }
+            $collection->addCompany($companyTransfer);
+        }
```
Put this logic to a mapper class.

daniel-rose commented on bundles/company-users-bulk-rest-api-debtor-connector/src/FondOfOryx/Zed/CompanyUsersBulkRestApiDebtorConnector/Persistence/CompanyUsersBulkRestApiDebtorConnectorRepository.php:
Too much sub queries. That is not performant.

daniel-rose commented on bundles/company-users-bulk-rest-api/composer.json:
```diff
@@ -0,0 +1,53 @@
+    "spryker/glue-application-extension": "^1.0.0",
+    "spryker/company-business-unit": "^1.0.0 || ^2.0.0",
+    "spryker/company-role": "^1.0.0",
+    "spryker/company-user": "^2.0.0",
+    "spryker/event": "^2.0.0",
```
Is version `1.0.0` not supported? And Why?

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Permission/PermissionChecker.php:
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\Permission;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\Checker;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/Permission/PermissionCheckerInterface.php:
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\Permission;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\Checker;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/PluginExecutioner/BulkDataPluginExecutioner.php:
```diff
@@ -0,0 +1,83 @@
+use Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer;
+use Generated\Shared\Transfer\RestCompanyUsersBulkResponseTransfer;
+
+class BulkDataPluginExecutioner implements BulkDataPluginExecutionerInterface
```
```suggestion
class BulkDataPluginExecutor implements BulkDataPluginExecutorInterface
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/PluginExecutioner/BulkDataPluginExecutioner.php:
```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\PluginExecutioner;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\Executor;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/PluginExecutioner/BulkDataPluginExecutionerInterface.php:
```diff
@@ -0,0 +1,33 @@
+use Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer;
+use Generated\Shared\Transfer\RestCompanyUsersBulkResponseTransfer;
+
+interface BulkDataPluginExecutionerInterface
```
```suggestion
interface BulkDataPluginExecutorInterface
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/PluginExecutioner/BulkDataPluginExecutionerInterface.php:
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\PluginExecutioner;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\Executor;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Communication/Plugin/CompanyResolverPreDataExpanderPlugin.php:
```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Plugin;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Plugin\CompanyUsersBulkRestApiExtension;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Communication/Plugin/CustomerByEmailResolverPreDataExpanderPlugin.php:
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Plugin;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Plugin\CompanyUsersBulkRestApiExtension;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Communication/Plugin/CustomerByReferenceResolverPreDataExpanderPlugin.php:
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Plugin;
```
```suggestion
namespace FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Plugin\CompanyUsersBulkRestApiExtension;
```

daniel-rose commented on bundles/company-users-bulk-rest-api/src/FondOfOryx/Zed/CompanyUsersBulkRestApi/Dependency/Facade/CompanyUsersBulkRestApiToCompanyUserFacadeBridge.php:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * @var \Spryker\Zed\CompanyUser\Business\CompanyUserFacadeInterface
+     */
+    protected $facade;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiClientTest.php:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * @var \FondOfOryx\Client\CompanyUsersBulkRestApi\CompanyUsersBulkRestApiFactory|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $factoryMock;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiClientTest.php:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * @var \FondOfOryx\Client\CompanyUsersBulkRestApi\Zed\CompanyUsersBulkRestApiStubInterface|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $zedStubMock;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiClientTest.php:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * @var \Generated\Shared\Transfer\RestCompanyUsersBulkRequestTransfer|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $restRequestTransferMock;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiClientTest.php:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * @var \Generated\Shared\Transfer\RestCompanyUsersBulkResponseTransfer|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $restResponseTransferMock;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiClientTest.php:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * @var \FondOfOryx\Client\CompanyUsersBulkRestApi\CompanyUsersBulkRestApiClient
+     */
+    protected $client;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiFactoryTest.php:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * @var \PHPUnit\Framework\MockObject\MockObject|\Spryker\Client\Kernel\Container
+     */
+    protected $containerMock;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiFactoryTest.php:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * @var \FondOfOryx\Client\CompanyUsersBulkRestApi\Dependency\Client\CompanyUsersBulkRestApiToZedRequestClientInterface|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $zedRequestClientMock;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiFactoryTest.php:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * @var \FondOfOryx\Client\CompanyUsersBulkRestApi\CompanyUsersBulkRestApiFactory
+     */
+    protected $factory;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/Dependency/Client/CompanyUsersBulkRestApiToZedRequestClientBridgeTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Client/CompanyUsersBulkRestApi/Zed/CompanyUsersBulkRestApiStubTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Glue/CompanyUsersBulkRestApi/CompanyUsersBulkRestApiFactoryTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Glue/CompanyUsersBulkRestApi/Controller/CompanyUsersBulkRestApiResourceControllerTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Glue/CompanyUsersBulkRestApi/Plugin/GlueApplicationExtension/CompanyUsersBulkResourceRoutePluginTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Glue/CompanyUsersBulkRestApi/Processor/Builder/RestResponseBuilderTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Glue/CompanyUsersBulkRestApi/Processor/CompanyUsersBulk/CompanyUsersBulkProcessorTest.php:
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Glue/CompanyUsersBulkRestApi/Processor/Mapper/RestCompanyUsersBulkRequestMapperTest.php:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * @var \FondOfOryx\Glue\CompanyUsersBulkRestApi\Processor\Mapper\RestCompanyUsersBulkRequestMapper
+     */
+    protected $mapper;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Zed/CompanyUsersBulkRestApi/Business/CompanyUsersBulkRestApiBusinessFactoryTest.php:
```diff
@@ -0,0 +1,137 @@
+    /**
+     * @var \FondOfOryx\Zed\CompanyUsersBulkRestApi\Business\CompanyUsersBulkRestApiBusinessFactory
+     */
+    protected $factory;
```
Use typed properties...

daniel-rose commented on bundles/company-users-bulk-rest-api/tests/FondOfOryx/Zed/CompanyUsersBulkRestApi/Communication/Controller/GatewayControllerTest.php:
```diff
@@ -0,0 +1,95 @@
+    /**
+     * @var \FondOfOryx\Zed\CompanyUsersBulkRestApi\Communication\Controller\GatewayController
+     */
+    protected $gatewayController;
```
Use typed properties...

daniel-rose reviewed (commented):
company-users-bulk-rest-api-debtor-connector -> business-central-company-users-bulk-rest-api or company-users-bulk-rest-api-business-central-connector

julianhyatt committed:
rename company-users-bulk-rest-api-debtor-connector => company-users-bulk-rest-api-business-central-connector

julianhyatt committed:
fix type properties

julianhyatt committed:
performance optimisation:
Select specific data from db instead of working with whole entities and prevent usage of from to array on entities

julianhyatt committed:
fix ci and add unit tests

julianhyatt committed:
Merge branch 'main' into feature/company-users-bulk-rest-api

julianhyatt committed:
remove local testing stuff

julianhyatt committed:
decouple logic

julianhyatt commented:
The integration is moved to https://github.com/fond-of-impala/
